File: src/01-easy/the-river-i-/solution.cs
using System;
class Solution {
    static int Ds(long n){int s=0;while(n>0){s+=(int)(n%10);n/=10;}return s;}
    static void Main(){
        long r1=long.Parse(Console.ReadLine()),r2=long.Parse(Console.ReadLine());
        while(r1!=r2){if(r1<r2)r1+=Ds(r1);else r2+=Ds(r2);}
        Console.WriteLine(r1);
    }
}

File: src/01-easy/the-river-i-/solution.vb
Imports System
Module Solution
    Function Ds(ByVal n As Long) As Long
        Dim s As Long = 0
        While n > 0
            s += n Mod 10
            n = n \ 10
        End While
        Return s
    End Function
    Sub Main()
        Dim r1 As Long = Long.Parse(Console.ReadLine())
        Dim r2 As Long = Long.Parse(Console.ReadLine())
        While r1 <> r2
            If r1 < r2 Then r1 += Ds(r1) Else r2 += Ds(r2)
        End While
        Console.WriteLine(r1)
    End Sub
End Module
